### PR TITLE
Mat 4508/disable measure group buttons until dirty form

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-4508](https://jira.cms.gov/browse/MAT-4508)
(Optional) Related Tickets:

### Summary

Disabling save and discard after they are clicked

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/components/measureGroups/MeasureGroups.tsx [hunk 1]
```diff
@@ -278,6 +278,7 @@ const MeasureGroups = () => {
           setSuccessMessage(
             "Population details for this group updated successfully."
           );
+          formik.resetForm();
```
Comment (RohitKandimalla) [inline src/components/measureGroups/MeasureGroups.tsx:281]: Why do we have to reset the form after a successful update?

Comment (RohitKandimalla) [reply src/components/measureGroups/MeasureGroups.tsx:281]: If it is required to do so, Do we also have to reset the form after creating a new group?

File: src/components/measureGroups/MeasureGroups.test.tsx [hunk 2]
```diff
@@ -399,21 +399,6 @@ describe("Measure Groups Page", () => {
     userEvent.click(
       screen.getByTestId("leftPanelMeasureInformation-MeasureGroup1")
     );
```
Comment (RohitKandimalla) [inline src/components/measureGroups/MeasureGroups.test.tsx]: Are we expecting any actions after the click event, if not is it required to trigger a click event?

File: src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx [hunk 3]
```diff
@@ -47,7 +47,7 @@ export default function MeasureDetailsSidebar(
 
   useEffect(() => {
     if (links) setMeasureGroups(links);
-  }, []);
+  }, [measure?.groups]);
```
Comment (RohitKandimalla) [inline src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx:50]: Is measure Group a dependency for this hook?
but it is never used inside the hook, so there is no impact of this dependency.

Comment (sb-prateekkeerthi) [reply src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx:50]: Yeah it is necessary dependency, this is for a scenario where you create measure group 1 and measure group 2 but don't save it but if you save measure group 3 then automatically this hook will set the information of MeasureGroups to updated Measure groups value as result of which measure group 1 and measure group 2 will not be shown it would be the measure group 3 which would be shown as measure group 1

File: src/components/measureGroups/MeasureGroups.test.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/measureGroups/MeasureGroups.test.tsx]: Sorry, I forgot to remove it. I'll remove it and will push it

File: src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx (hunk 3 above)
Comment (RohitKandimalla) [reply src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx:50]: Do we want to do that? Is it a requirement?

Comment (sb-prateekkeerthi) [reply src/components/editMeasure/measureDetails/MeasureDetailsSidebar.tsx:50]: Yeah, for now we are implementing it, but going further when we implement the validations like: you can't create a new group until you save previous one in that case we can just remove it

File: src/components/measureGroups/MeasureGroups.tsx (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/components/measureGroups/MeasureGroups.tsx:281]: The reset form is basically for checking if the form is dirty and depending on which we are disabling the save and discard buttons

Comment (sb-prateekkeerthi) [reply src/components/measureGroups/MeasureGroups.tsx:281]: In case of creating the group the form starts with an empty one so you need not use it, but where as in case of update you all ready have the previous values so that is the reason you use it in update for dirty check and validity

**Review by RohitKandimalla — APPROVED** [review]
